paddle/fluid/platform/mkldnn_reuse.h

```diff
@@ -395,9 +396,28 @@ class TransposeMKLDNNHandler : public MKLDNNHandler {
   std::vector<int> logical_axis_;
 };
 
+template <typename T>
+struct mode;
```

Comment:
what is mode? seems like `mkldnn::algorithm`

Reply:
@tensor-tang Yes, mode was introduced so that When AcquireConvolutionPrimitiveDescriptor
is working for conv_mkldnn eg. template parameter is "convolution_forward"  then mkldnn::algorithm should be "convolution_direct" . When AcquireConvolutionPrimitiveDescriptor
is working for conv_transpose_mkldnn eg. template parameter is "deconvolution_forward" then mkldnn::algorithm should be "desconvolution_direct" . So mode was introduced to have this mapping done